server/src/main/java/com/google/coffeehouse/servlets/GetProfileServlet.java

```diff
@@ -0,0 +1,102 @@
+ */
+@WebServlet("/api/get-profile")
+public class GetProfileServlet extends HttpServlet {
+
+  /** 
+   * The error string sent by the response object in doGet when the body of the 
+   * GET request cannot be used to fetch a {@link Person} for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+
+  /** The logged error string when an error parsing the body of the GET request is encountered. */
+  public static final String LOG_BODY_ERROR_MESSAGE = 
+      "Body unable to be parsed in GetProfileServlet: ";
+
+  /** Message to be logged when the body of the GET request does not have required fields. */
+  public static final String LOG_INPUT_ERROR_MESSAGE =
+      "Error with JSON input in GetProfileServlet: No \"userId\" found in JSON.";
+
+  private static final Gson gson = new Gson();
+  private final StorageHandlerApi storageHandler;
+
+  /**
+   * Overloaded constructor for dependency injection.
+   * @param storageHandler the {@link StorageHandlerApi} that is used when fetching the Person
+   */
+  public GetProfileServlet(StorageHandlerApi storageHandler) {
+    super();
+    this.storageHandler = new StorageHandlerApi();
+  }
+
+  /**
+   * Explicity default constructor used for instantiating the servlet when not testing.
+   */
+  public GetProfileServlet() {
+    super();
+    this.storageHandler = new StorageHandlerApi();
+  }
+  
+  /** 
+   * Returns a {@link Person} object in JSON format from information in the database.
+   * @param request the GET request that must have a valid JSON representation of the userId to be
+   *     passed in order to fetch a person from ID in the database. If this is not the case the
+   *     response will send a "400 Bad Request error"
+   * @param response the response from this method, will contain the object in JSON format.
+   *     If the request object does not have a valid JSON body that describes the Person, this object
+   *     will send a "400 Bad Request error"
+   * @throws IOException if an input or output error is detected when the servlet handles the request
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Person person;
+    try {
+      Map userInfo = gson.fromJson(request.getReader(), Map.class);
+      // TODO: Change to actual oauth parameter from json that is passed in @JoeyBushagour.
+      String userId = (String) userInfo.get(Person.USER_ID_FIELD_NAME);
+      if (userId == null) {
+        throw new IllegalArgumentException(LOG_INPUT_ERROR_MESSAGE);
+      }
+      person = storageHandler.fetchPersonFromId(userId);
+    } catch (IllegalArgumentException e) {
+      System.out.println(e.getMessage());
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
+      return;
+    } catch (Exception e) {
+      System.out.println(LOG_BODY_ERROR_MESSAGE + e.getMessage());
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, BODY_ERROR);
```

Comment:
It's generally good to vary these error codes up and make sure they're semantically correct (eg. they correspond to what actually happened). This one would probably be better as HttpServletResponse.SC_INTERNAL_SERVER_ERROR

Reply:
It doesn't look like this was updated.

Reply:
It was updated, this comment is now on an outdated version of the file. I believe if you navigate to Files Changed, this error was fixed.